File: account/src/main/java/com/vzk/account/models/Team.java
package com.vzk.account.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;
import java.util.UUID;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "teams")
public class Team {
    @Id
    @Column(name = "id")
    private UUID id;

    @Basic
    @Column(name = "name")
    private String name;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "captain", referencedColumnName = "id")
    private Account captain;

    @ManyToOne
    @JoinColumn(name = "game")
    private Game game;

    @Basic
    @Column(name = "description")
    private String description;

    @Basic
    @Column(name = "is_active")
    private boolean isActive;
}

File: ui/vzk-ui/src/app/models/team/team.ts
import {Game} from "../game/game";
import {Player} from "../player/player";

export class Team{
  id: string;
  name: string;
  game: Game;
  players: Player[];
  description: string;


  constructor(id: string, name: string, game: Game, players: Player[], description: string) {
    this.id = id;
    this.name = name;
    this.game = game;
    this.players = players;
    this.description = description;
  }
}
